host/xtest/regression_8100.c

```diff
@@ -0,0 +1,316 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright (c) 2018, Linaro Limited */
+
+#include "xtest_test.h"
+#include "xtest_helpers.h"
+
+#include <compiler.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ta_crypt.h>
+#include <tee_api_types.h>
+#include <adbg.h>
+
+#ifdef OPENSSL_FOUND
+#include <openssl/x509_vfy.h>
+#include <openssl/pem.h>
+#include <openssl/err.h>
+#include <openssl/crypto.h>
+#endif
+
+#include "regression_8100_ca_crt.h"
+#include "regression_8100_mid_crt.h"
+#include "regression_8100_my_crt.h"
+#include "regression_8100_my_csr.h"
+
+#ifdef CFG_TA_MBEDTLS
+
+static void test_8101(ADBG_Case_t *c __maybe_unused)
+{
+#ifdef CFG_TA_MBEDTLS_SELF_TEST
+	TEEC_Session session = { 0 };
+	uint32_t ret_orig;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, xtest_teec_open_session(
+					      &session, &crypt_user_ta_uuid,
+					      NULL, &ret_orig)))
+		return;
+	ADBG_EXPECT_TEEC_SUCCESS(c,
+		TEEC_InvokeCommand(&session, TA_CRYPT_CMD_MBED_SELF_TESTS, NULL,
+				   &ret_orig));
+	TEEC_CloseSession(&session);
+#else
+	Do_ADBG_Log("CFG_TA_MBEDTLS_SELF_TEST not set, test skipped");
+#endif
+}
+ADBG_CASE_DEFINE(regression, 8101, test_8101, "TA mbedTLS self tests");
+
+static int __printf(2, 3) myasprintf(char **strp, const char *fmt, ...)
```

Comment:
Why not use `asprintf()` from `stdio.h>`? Android I suppose?

Reply:
It was actually missing in the libc provided by buildroot too.